src/test/java/BaseTest.java

```diff
@@ -1,18 +1,172 @@
     @BeforeSuite
     static void setupClass() {
-        WebDriverManager.chromedriver().setup();
+//        WebDriverManager.chromedriver().setup();
+    }
+
+    @BeforeMethod
+    @Parameters({"BaseURL"})
+    public static void launchBrowser(String BaseURL) throws MalformedURLException {
+        driver = pickBrowser(System.getProperty("browser"));
+//        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));
+        url = BaseURL;
+        driver.get(url);
+        wait = new WebDriverWait(driver, Duration.ofSeconds(10));
+    }
+
+
+    public static WebDriver pickBrowser(String browser) throws MalformedURLException {
+        DesiredCapabilities caps = new DesiredCapabilities();
+        String gridURL = "http://192.168.1.10:4444";
+
+        switch (browser){
+            case "firefox":
+                WebDriverManager.firefoxdriver().setup();
+                driver = new FirefoxDriver();
+                break;
+            case "MicrosoftEdge":
+                WebDriverManager.edgedriver().setup();
+                driver = new EdgeDriver();
+                break;
+            case "grid-edge":
+                caps.setCapability("browserName", "MicrosoftEdge");
+                driver = new RemoteWebDriver(URI.create(gridURL).toURL(), caps);
+                break;
+            case "grid-firefox":
+                caps.setCapability("browserName", "FireFox");
+                driver = new RemoteWebDriver(URI.create(gridURL).toURL(), caps);
+                break;
+            case "grid-chrome":
+                caps.setCapability("browserName", "chrome");
+                driver = new RemoteWebDriver(URI.create(gridURL).toURL(), caps);
+                break;
+            default:
+                WebDriverManager.chromedriver().setup();
+                driver = new ChromeDriver();
+        }
+        return driver;
+    }
+
+    @AfterMethod
+    public static void closeBrowser() {
+        driver.quit();
+    }
+
+//    public static void login(String email, String password){
+//        provideEmail(email);
+//        providePassword(password);
+//        clickSubmit();
+//    }
+
+//    public static void navigateToPage() {
+//        String url = "https://bbb.testpro.io/";
+//        driver.get(url);
+//    }
+
+    @DataProvider(name="IncorrectLoginProvider")
+    public static Object[][] getDataFromDataProviders(){
+        return new Object[][] {
+                {"invalid@email.com", "invalidPas"},
+                {"demo@mail.com", "invalid"},
+                {"", ""}
+        };
     }
+//    public static void provideEmail(String email) {
+//        WebElement emailField = wait.until(ExpectedConditions.elementToBeClickable(By.cssSelector("[type='email']")));
+//
+//        emailField.clear();
+//        emailField.sendKeys(email);
+//    }
+//    public static void providePassword(String password) {
+//        WebElement passwordField = driver.findElement(By.cssSelector("[type='password']"));
+//
```

Comment:
**I recommend not using code comments**: for, example, when you have a code that cannot be understood, then comments are not the solution. Try to rewrite the code or rename fields and other elements in a way that the reader will understand the action that you are doing there.